### PR TITLE
Add tool for codegen removal

### PR DESCRIPTION
**Pull request by kilativ-dotcom**
* [X] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/CONTRIBUTING.md)
* [ ] Update changelog
* [X] Update documentation

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@ nav:
       - Knowledge Base repo file: sc-tools/kb_repo_file.md
       - Websocket Server: sc-tools/sc_server.md
       - Sc-machine Runner: sc-tools/sc_machine.md
+      - Codegen removal: sc-tools/codegen_removal.md
```
Comment (NikitaZotov) [inline mkdocs.yml:48]: I think we should emphasise that this is a script to migrate from the old API to the new one. I propose to name this script `migrate_to_new_agent_api` and note that it replaces the old API using codegeneration with a new API

File: scripts/codegen-removal/remove_codegen.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+set -eo pipefail
+CURRENT_DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd)"
+
+java -jar "$CURRENT_DIR/codegen-removal.jar" "$@"
```
Comment (NikitaZotov) [inline scripts/codegen-removal/remove_codegen.sh:5]: Please, check that java exists. If it isn't installed, then write help, that user should install java8

File: docs/sc-tools/codegen_removal.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# codegen-removal
+
+This is a program that translates sc-machine usage from old format into new format.
+
+to run you can use script
+```shell
+./scripts/codegen-removal/remove_codegen.sh /absolute/path/to/folder/with/project
+```
+
+there may be a situation when this program won't be able to replace code correctly, these places are marked with `//todo(codegen-removal):`
```
Comment (NikitaZotov) [inline docs/sc-tools/codegen_removal.md:10]:
```suggestion
There may be a situation when this program won't be able to replace code correctly, these code places are marked with `//todo(codegen-removal):`
```

File: docs/sc-tools/codegen_removal.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# codegen-removal
+
+This is a program that translates sc-machine usage from old format into new format.
```
Comment (NikitaZotov) [inline docs/sc-tools/codegen_removal.md:3]:
```suggestion
This is a program that translates all agents implemented on the old API supported on sc-machine versions <= 0.9.0 to the new API.
```

File: docs/sc-tools/codegen_removal.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+to run you can use script
```
Comment (NikitaZotov) [inline docs/sc-tools/codegen_removal.md:5]:
```suggestion
To run you can use this script
```

File: docs/sc-tools/codegen_removal.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+```shell
+./scripts/codegen-removal/remove_codegen.sh /absolute/path/to/folder/with/project
```
Comment (NikitaZotov) [inline docs/sc-tools/codegen_removal.md:7]: It can be relative path?

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: docs/sc-tools/codegen_removal.md (hunk 4 above)
Comment (kilativ-dotcom) [reply docs/sc-tools/codegen_removal.md:3]: not only agents are translated, I added list with everything that is translated

File: docs/sc-tools/codegen_removal.md (hunk 6 above)
Comment (kilativ-dotcom) [reply docs/sc-tools/codegen_removal.md:7]: no it cannot be a relative path

Comment (NikitaZotov) [reply docs/sc-tools/codegen_removal.md]: why?

File: docs/index.md [hunk 7]
```diff
@@ -36,6 +36,7 @@
     * [Knowledge Base repo file](sc-tools/kb_repo_file.md) - *description of configuration of knowledge base sources*
     * [Websocket Server](sc-tools/sc_server.md) - *description of options of sc-server for communication with sc-memory through network*
     * [Sc-machine Runner](sc-tools/sc_machine.md) - *description of options of sc-machine runner*
+    * [Migrate to new agent api](sc-tools/migrate_to_new_agent_api.md) - *description of script that can be used to migrate to new C++ Agents API*
```
Comment (NikitaZotov) [inline docs/index.md:39]:
```suggestion
    * [Migrate to new agent API](sc-tools/migrate_to_new_agent_api.md) - *description of script that can be used to migrate to new C++ Agents API*
```

File: scripts/migration/migrate_to_new_agent_api.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+set -eo pipefail
+CURRENT_DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd)"
+source "${CURRENT_DIR}/../formats.sh"
+
+if ! which java &> /dev/null; then
+    warning "Java is not installed. To run this you need java with version 8 or higher installed."
+    exit 1
+fi
+
+java -jar "$CURRENT_DIR/migrate_to_new_agent_api.jar" "$@"
```
Comment (NikitaZotov) [inline scripts/migration/migrate_to_new_agent_api.sh:11]:
```suggestion
java -jar "${CURRENT_DIR}/migrate_to_new_agent_api.jar" "$@"
```

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: docs/sc-tools/codegen_removal.md (hunk 6 above)
Comment (kilativ-dotcom) [reply docs/sc-tools/codegen_removal.md]: script may be invoked from anywhere and relative path for that place may differ from relative path for script location and it is not necessary to support that

Comment (NikitaZotov) [reply docs/sc-tools/codegen_removal.md]: I think support for the relative path should be made. We support it everywhere

Comment (kilativ-dotcom) [reply docs/sc-tools/codegen_removal.md]: image
not everywhere

Comment (kilativ-dotcom) [reply docs/sc-tools/codegen_removal.md]: turns out it did work with not absolute paths

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**